src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    if (this.health > 0) {
+      Animal.alive.push(this);
+    } else {
+      throw new Error('Animals must be alive');
+    }
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden) {
+    super(name, health);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivore) {
+    // I suppose I can't do (!herbivore.hidden) like mentor noticed
+    // because I should check is it false, but not undefined
```

Comment:
Hmm, ok, it's tricky case as we don't have specific checkpoints about it in our task, so let it be then👍
But on next time don't write comments in code - ask your questions in the Slack